src/main/java/edu/hm/hafner/analysis/descriptor/Descriptor.java

```diff
@@ -0,0 +1,39 @@
+package edu.hm.hafner.analysis.descriptor;
+
+import edu.hm.hafner.analysis.IssueParser;
+
+public interface Descriptor {
```

Comment:
Using an interface in an API is hard to evolve: use an abstract base class. Then we can also provide empty default implementations for pattern and other optional properties. (This can be done as refactoring in IntelliJ so you don't need to manually remove the empty strings in the concrete classes.) Maybe we should also name it `ParserDescriptor` or `ToolDescriptor`